HeartbeatLogging/Sources/HeartbeatStorage.swift

```diff
@@ -48,29 +61,50 @@ final class HeartbeatStorage: HeartbeatStorageProtocol {
 
   // MARK: - Instance Management
 
-  /// <#Description#>
+  /// Statically allocated cache of `HeartbeatStorage` instances.
   private static var cachedInstances: [String: WeakContainer<HeartbeatStorage>] = [:]
 
-  /// <#Description#>
-  /// - Parameter id: <#id description#>
-  /// - Returns: <#description#>
+  /// Gets an existing `HeartbeatStorage` instance with the given `id` if one exists. Otherwise,
+  /// makes a new instance with the given `id`.
+  ///
+  /// - Parameter id: A string identifier.
+  /// - Returns: A `HeartbeatStorage` instance.
   static func getInstance(id: String) -> HeartbeatStorage {
     if let cachedInstance = cachedInstances[id]?.object {
       return cachedInstance
     } else {
-      let newInstance = HeartbeatStorage.makeStorage(id: id)
+      let newInstance = HeartbeatStorage.makeHeartbeatStorage(id: id)
       cachedInstances[id] = WeakContainer(object: newInstance)
       return newInstance
     }
   }
 
+  /// Makes a `HeartbeatStorage` instance using a given `String` identifier.
+  ///
+  /// The created persistent storage object is platform dependent. For tvOS, user defaults
+  /// is used as the underlying storage container due to system storage limits. For all other platforms,
+  /// the file system is used.
+  ///
+  /// - Parameter id: A `String` identifier used to create the `HeartbeatStorage`.
+  /// - Returns: A `HeartbeatStorage` instance.
+  private static func makeHeartbeatStorage(id: String) -> HeartbeatStorage {
+    #if os(tvOS)
+      let storage = UserDefaultsStorage.makeStorage(id: id)
+    #else
+      let storage = FileStorage.makeStorage(id: id)
+    #endif // os(tvOS)
+    return HeartbeatStorage(id: id, storage: storage)
+  }
+
   deinit {
     // Removes the instance if it was cached.
     Self.cachedInstances.removeValue(forKey: id)
   }
 
   // MARK: - HeartbeatStorageProtocol
 
+  /// Asynchronously reads from and writes to storage using the given transform block.
+  /// - Parameter transform: A block to transform `HeartbeatInfo?` to `HeartbeatInfo?`.
```

Comment:
transform `HeartbeatInfo?` to `HeartbeatInfo?` ?

Reply:
Take a look at this [piece of documentation](https://github.com/firebase/firebase-ios-sdk/blob/60a39cb6cea4daceada1a50566f1226b4e7bf2d1/HeartbeatLogging/Sources/HeartbeatStorage.swift#L17-L23) which I added in this PR to explain transformation. 

PTAL and LMK what you think. Im open to suggestions

Reply:
If you agree with the idea behind the `HeartbeatInfo?` → `HeartbeatInfo?` semantics, then maybe this doc could instead read something like:
```suggestion
  /// - Parameter transform: A block that transforms the current stored value to a new value.
```

Reply:
Maybe I'm missing too much context, but when I see the word transform, I expect to see an explanation of the before state and the after state. Instead I see two exactly matching strings and nothing that describes what was transformed or why it was transformed. And the documentation link talks about managing nil, but not about the transformation that I can tell.

Reply:
I see. I'll schedule some then to discuss because I'd like the intent to be clear.